patrick-lab/server.js

```diff
@@ -0,0 +1,73 @@
+'use strict';
+
+const Client = require('./model/pages.js');
+const net = require('net');
+const EE = require('events').EventEmitter;
+const ee = new EE();
+const server = net.createServer();
+const PORT = process.env.PORT || 3000;
+
+const pool = [];
+
+ee.on('default', (client, string) => {
+  client.socket.write(`Not a valid command: ${string.split(' ', 1)}\n`);
+});
+
+ee.on('/all', (client, string) => {
+  pool.forEach(c => c.socket.write(`${client.nickname}: ${string}`));
+});
+
+ee.on('/nick', (client, string) => {
+  client.nickname = string.trim();
```

Comment:
No points lost for this, but you should do a ``client.socket.write`` here that indicates the user successfully changed their nickname for a better UX.